usersession/agent/rest_api.go

```diff
@@ -38,6 +46,11 @@ var (
 		Path: "/v1/session-info",
 		GET:  sessionInfo,
 	}
+
+	servicesCmd = &Command{
+		Path: "/v1/services",
+		POST: postServices,
```

Comment:
Do we need any kind of authentication here? Do we need to enforce that the user making the request is the same as the user running the user session daemon?

Reply:
The intended audience for this API is root.  Other users should not be able to access the the session agent due to the permissions on `$XDG_RUNTIME_DIR`.

On top of this, PR #7197 adds a second layer of protection by performing an `SO_PEERCRED` check on new connections.  If we're only performing actions on behalf of the same user or root, we don't need the complication of snapd's per-user authentication checks.

Reply:
FWIW, PR 7555 addresses my concerns about 'another line of defense' since with strict mode, apparmor will ensure snaps can't interfere with the socket. For non-snaps, the socket is protected via DAC so other non-root users can't access it and for the session user, the API provides no additional privilege. What is left is partial confinement systems where a snap might have access to the socket [due to a lack of unix mediation patches](https://github.com/snapcore/snapd/pull/7555#discussion_r354452340). From my pov, that is not a blocker, but there is an opportunity for a future hardening enhancement.